eslint.config.mjs

```diff
@@ -123,6 +125,10 @@ export default tseslint.config([
 
       'no-floating-promise/no-floating-promise': 'error',
 
+      // Use the recommended rules for react-hooks
+      ...reactHooks.configs['recommended-latest'].rules,
+      'react-hooks/exhaustive-deps': 'error',
```

Comment:
Nit: this plugin only offers two rules. Consider just inlining them both.

https://github.com/facebook/react/blob/a7a116577daf3b135c226ed9db8a8c2f9166c023/packages/eslint-plugin-react-hooks/src/index.ts#L22-L23


```suggestion
      'react-hooks/rules-of-hooks': 'error',
      'react-hooks/exhaustive-deps': 'error',
```

If you still want to use the preset, don't use `recommended-latest`, it's deprecated:

https://github.com/facebook/react/blob/a7a116577daf3b135c226ed9db8a8c2f9166c023/packages/eslint-plugin-react-hooks/src/index.ts#L55-L56